app/model/repository/ReferenceSolutionSubmissions.php

```diff
@@ -20,4 +14,20 @@ public function __construct(EntityManager $em) {
     parent::__construct($em, ReferenceSolutionSubmission::class);
   }
 
+  /**
+   * Find all submissions created in given time interval.
+   * @param DateTime|null $since Only submissions created after this date are returned.
+   * @param DateTime|null $until Only submissions created before this date are returned.
+   */
+  public function findByCreatedAt(?DateTime $since, ?DateTime $until)
+  {
+    $qb = $this->createQueryBuilder('s'); // takes care of softdelete cases
```

Comment:
You wrote similar function three times in this PR, maybe probably some helper function on `BaseRepository` can be introduced?